xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxFFmpeg.cpp

```diff
@@ -655,9 +655,26 @@ AVDictionary *CDVDDemuxFFmpeg::GetFFMpegOptionsFromInput()
         av_dict_set(&options, "user-agent", value.c_str(), 0);
         hasUserAgent = true;
       }
+      // other standard headers (see https://en.wikipedia.org/wiki/List_of_HTTP_header_fields)
+      else if (name == "accept" || name == "accept-language" || name == "accept-datetime" || 
+        name == "authorization" || name == "cache-control" || name == "connection" || name == "cookie" || name == "content-md5" || 
+        name == "content-type" || name == "date" || name == "expect" || name == "forwarded" || name == "from" || name == "host" || name == "if-match" || 
+        name == "if-modified-since" || name == "if-none-match" || name == "if-range" || name == "if-unmodified-since" || name == "max-forwards" || 
+        name == "origin" || name == "pragma" || name == "proxy-authorization" || name == "range" || name == "referer" || name == "te" || name == "upgrade" || 
+        name == "via" || name == "warning" || name == "x-requested-with" || name == "dnt" || name == "x-forwarded-for" || name == "x-forwarded-host" || 
+        name == "x-forwarded-proto" || name == "front-end-https" || name == "x-http-method-override" || name == "x-att-deviceid" || 
+        name == "x-wap-profile" || name == "proxy-connection" || name == "x-uidh" || name == "x-csrf-token" || name == "x-request-id" || name == "x-correlation-id")
+      {
+        CLog::Log(LOGDEBUG, "CDVDDemuxFFmpeg::GetFFMpegOptionsFromInput() adding custom header option '%s: %s'", it->first.c_str(), value.c_str());
+        headers.append(it->first).append(": ").append(value).append("\r\n");
```

Comment:
ok, that should to. ffmpeg seems to treat headers as if it were set by option directly.